Sources/NIO/IntegerTypes.swift

```diff
@@ -75,7 +75,8 @@ struct _UInt56 {
 
     static let bitWidth: Int = 56
 
-    static let max: _UInt56 = .init((1 << 56) - 1)
+    private static let initializeUInt64 : UInt64 = (1 << 56) - 1
+    static let max: _UInt56 = .init(initializeUInt64)
```

Comment:
I think the problem here is that this numeric literal is always typed as `Int32` on 32-bit platforms, even when I set this with `.init( UInt64 ( (1 << 56) - 1 ) )`, which then overflows the `Int32` and causes a segfault. Defining a separate `UInt64` variable works around this, but perhaps there's a better way to do it?

Reply:
I think the fact that your proposed workaround didn't work seems like a bug, and that should be filed. This is an appropriate fix, given that your original attempt didn't work.